code/modules/projectiles/projectile.dm

```diff
@@ -1052,6 +1052,14 @@
 				if(!stat && pain.feels_pain)
 					emote("scream")
 					to_chat(src, SPAN_HIGHDANGER("You scream in pain as the impact sends <B>shrapnel</b> into the wound!"))
+
+	var/mob/shoot_mob = P.weapon_cause_data.resolve_mob()
```

Comment:
Can `weapon_cause_data` be null? I think yes: (most definitely a bug, since the first argument is location, but still)
https://github.com/cmss13-devs/cmss13/blob/0cbe3e21446a2af36c317845afbc5fdd9c208ee8/code/modules/defenses/sentry_flamer.dm#L32
I think it would be better to always check if `weapon_cause_data == null` and log it somewhere
With current code any projectile without weapon_cause_data will runtime and not deal any damage
```suggestion
	var/mob/shoot_mob = P.weapon_cause_data?.resolve_mob()
```

Reply:
bit too late lol